src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * <p>
+ * Checks if the javadoc has
+ * <a href="https://docs.oracle.com/en/java/javase/14/docs/specs/javadoc/doc-comment-spec.html#leading-asterisks">
+ * leading asterisk
+ * </a> on each line.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingLeadingAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;** // violation, javadoc has lines without leading asterisk.
+ *  * Some description here.
+ *    Another line of description.
+ *  *&#47;
+ * public class Test {
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *    * Another line of description.
+ *      Another line of description.
+ *    *&#47;
+ *   public void test() {}
+ *
+ *   &#47;** Some description here. *&#47; // OK
+ *   public void test1() {}
+ *
+ *   &#47;** // OK
+ *    * Some description here.
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test2() {}
+ *
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test3() {}
+ * }
+ * </pre>
+ *
+ * @since 8.33
+ */
+@StatelessCheck
+public class MissingLeadingAsteriskCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    /**
+     * The new line character that separates each line of javadoc.
+     */
+    private static final Pattern MATCH_COMMENT_SEPARATOR = CommonUtil.createPattern("\r\n?|\n");
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final String commentContent = JavadocUtil.getBlockCommentContent(ast);
+        if (JavadocUtil.isJavadocComment(commentContent)) {
+            final String[] commentLines = MATCH_COMMENT_SEPARATOR.split(commentContent);
+
+            for (int i = 0; i < commentLines.length - 1; i++) {
+                final String commentLine = commentLines[i];
+                if (!isValidLine(commentLine)) {
+                    log(ast, MSG_MISSING_ASTERISK);
+                }
+            }
+
+            // To validate the last line of javadoc
+            final String lastCommentLine = commentLines[commentLines.length - 1];
+            if (!CommonUtil.isBlank(lastCommentLine) && lastCommentLine.trim().charAt(0) != '*') {
+                log(ast, MSG_MISSING_ASTERISK);
```

Comment:
please show me what UT and line in Input covers this case.

> commentLines[commentLines.length - 1];

This might be a problem on empty comment, please also share link to Input that covers it, and explain why it is not failing.

Reply:
> please show me what UT and line in Input cover this case.

Here's [link](), please show me what UT and line in Input cover this case. 

> `commentLines[commentLines.length - 1];`
This might be a problem on empty comment, please also share link to Input that covers it, and explain why it is not failing.

Here's the [link]() to an empty comment input to the UT. As you can see in the input it has the following empty comment:
```
/***/
```
While parsing we get `"*"` as the content of Javadoc. The below statement then split's this string by new lines:
```
final String[] commentLines = MATCH_COMMENT_SEPARATOR.split(commentContent);
```
As there are no new lines present in the content, it returns `["*"]` as the result. The length of array is one due to which the for loop is skipped and finally the below condition, verifies that the string is not blank and it starts with a `*`:
```
if (!CommonUtil.isBlank(lastCommentLine) && lastCommentLine.trim().charAt(0) != '*') {
```
which is true, the string contains only `*`.

Hence, our code is not failing for empty comments.